docs/query.md

```diff
@@ -0,0 +1,265 @@
+| `fork:`      | `f:`    | `yes` or `no`          | Filters forked repositories.                               | `fork:no`                              |
+| `lang:`      | `l:`    | Text                   | Filters by programming language.                           | `lang:python`                          |
+| `public:`    |         | `yes` or `no`          | Filters public repositories.                               | `public:yes`                           |
+| `regex:`     |         | Regex pattern          | Matches content using a regular expression.                | `regex:/foo.*bar/`                     |
+| `repo:`      | `r:`    | Text (string or regex) | Filters repositories by name.                              | `repo:"github.com/user/project"`       |
+| `sym:`       |         | Text                   | Searches for symbol names.                                 | `sym:"MyFunction"`                     |
+| `branch:`    | `b:`    | Text                   | Searches within a specific branch.                         | `branch:main`                          |
+| `type:`      | `t:`    | `filematch`, `filename`, `file`, or `repo` | Limits result types.                   | `type:filematch`                       |
+
+---
+
+### 2. **Negation**
+
+Negate an expression using the `-` symbol.
+
+#### Examples:
+- Exclude a repository:
+  ```plaintext
+  -repo:"github.com/example/repo"
+  ```
+- Exclude a language:
+  ```plaintext
+  -lang:javascript
+  ```
+
+---
+
+### 3. **Grouping**
+
+Group queries using parentheses `()` to create complex logic.
+
+#### Examples:
+- Match either of two repositories:
+  ```plaintext
+  (repo:repo1 or repo:repo2)
+  ```
+- Find test in either python or javascript files:
+  ```plaintext
+  content:test (lang:python or lang:javascript)
+  ```
+
+---
+
+### 4. **Logical Operators**
+
+Use `or` to combine multiple expressions.
+
+#### Examples:
+- Match files in either of two languages:
+  ```plaintext
+  lang:go or lang:java
+  ```
+
+`and` boolean operator is applied automatically when expressions are separated by a space.
+
+---
+
+## Special Query Values
+
+- **Boolean Values**:
+  Use `yes` or `no` for fields like `archived:` or `fork:`.
+
+- **Text Fields**:
+  Text fields (`content:`, `repo:`, etc.) accept:
+  - Strings: `"my text"`
+  - Regular expressions: `/my.*regex/`
+
+- **Escape Characters**:
+  To include special characters, use backslashes (`\`).
+
+#### Examples:
+- Match the string `foo"bar`:
+  ```plaintext
+  content:"foo\"bar"
+  ```
+- Match the regex `foo.*bar`:
+  ```plaintext
+  content:/foo.*bar/
+  ```
+
+---
+
+## Advanced Examples
+
+1. **Search for content in Python files in public repositories**:
+   ```plaintext
+   lang:python public:yes content:"my_function"
+   ```
+
+2. **Exclude archived repositories and match a regex**:
+   ```plaintext
+   archived:no regex:/error.*handler/
```

Comment:
same nit on `/` not being needed for regular expressions. Also the `regex:` prefix is not needed since everything is matched as regex anyways

Reply:
You need / or regex if you want to do a regex with spaces in it.